src/Rule/AtLeast.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule;
+
+use Yiisoft\Validator\Rule;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\DataSetInterface;
+
+/**
+ * AtLeastValidator checks if at least $min of many attributes are filled.
+ */
+class AtLeast extends Rule
+{
+    /**
+     * The minimum required quantity of filled attributes to pass the validation.
+     * Defaults to 1.
+     */
+    private int $min = 1;
+
+    /**
+     * The list of required attributes that will be checked.
+     */
+    private array $attributes = [];
+
+    /**
+     * The list of alternative required attributes that will be checked.
+     */
+    private array $alternativeAttributes = [];
+
+    /**
+     * Message to display in case of error
+     */
+    private string $message = 'The model is not valid. Must have at least "{min}" filled attributes.';
+
+    public function __construct(array $data)
+    {
+        foreach ($data as $name => $value) {
```

Comment:
I think you should write `$this->attributes = $data` instead

Reply:
I agree. Allowing the constructor to set any properties(even the parent) via an array is not good.

```php
public function __construct(array $attributes, array $alternativeAttributes = [], int $min = 1)
```

Reply:
I updated the PR :)

Reply:
I'm sorry, I misled you. It is better to do it by analogy with other rules:

```php

    public function attributes(array $attributes): self
    {
        $new = clone $this;
        $new->attributes = $attributes;
        return $new;
    }

    public function min(int $min): self
    {
        $new = clone $this;
        $new->min = $min;
        return $new;
    }

```

And remove the constructor

It is also better to add the message method so that you can change the error text.